index.js

```diff
@@ -39,6 +40,17 @@ class RoutesFactory {
   }
 }
 
+const prepareMetadata = error => {
+  if(!error.metadata) {
+    return error;
+  }
+  const { metadata } = error;
+  const m = new Metadata();
+  Object.keys(error.metadata).forEach(key => m.add(key, String(metadata[key])));
+  error.metadata = m;
```

Comment:
I think it is better to create a new error object rather than override `error.metadata`, because `prepareMetadata` is  called every before sending response. Initially, error.metadata is a plane object, but after 1st API call, error.metadata will be a `grpc.Metadata` object. This could violate idempotency of mock handler.